### PR TITLE
Refactor initStaticDeps to return an error instead of panic

### PR DESCRIPTION
**Pull request by justinsb**
panic is generally a code-smell, IMO.

```release-note
NONE
```

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -65,7 +66,9 @@ func main() {
 	}
 
 	log.Printf("Initializing Dependencies.")
-	initStaticDeps(mgr)
+	if err := initStaticDeps(mgr); err != nil {
+		log.Fatal(err)
```
Comment (roberthbailey) [inline cmd/manager/main.go]: should this be klog?

Comment (BenTheElder) [reply cmd/manager/main.go]: all logging in main appears to be `log` rather than `klog` currently... 🤔 

@justinsb perhaps we should change this?

**Comment by BenTheElder** [comment]
> How does moving the fatal call one level up the stack prevent panic? Is it that klog panics and log doesn't?

IMHO, it's just that the only `exit()`-equivalent calls should occur in `main()`, that way the rest is re-usable, testable, etc, and only the actual application binary ever calls exit / panic / fatal / ...

**Comment by detiber** [comment]
/cancel-lifecycle rotten
/hold cancel

This looks fine, any additional changes can be addressed through followup (re klog vs log)

**Pull request merged**